### PR TITLE
[APM-1100] DMN service added

### PR DESCRIPTION
**Pull request by dhrn**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)
**What kind of change does this PR introduce?** (check one with "x")

This pr helps the extension to test decision table

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/ama-sdk/src/lib/api-implementations/acm-api/dmn-api.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * limitations under the License.
+ */
+
+import { Observable, from } from 'rxjs';
+import { Model } from '../../api/types';
+import { ModelApiInterface } from '../../api/generalmodel-api.interface';
+import { RequestApiHelper, RequestApiHelperOptions } from './request-api.helper';
+ import { ModelApiVariation } from '../acm-api/model-api';
+
+export class DmnApi<T extends Model, S> implements ModelApiInterface<T, S> {
```
Comment (popovicsandras) [inline projects/ama-sdk/src/lib/api-implementations/acm-api/dmn-api.ts]: Why do we need this new API? Isn't it the same as decision tables API? Or part of it?

File: projects/ama-sdk/src/lib/api-implementations/acm-api/dmn-api.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            bodyParam: {
+                'name' : tableName,
+                'input' : variables,
+                'xml': this.modelVariation.serialize(dmnTable)
+            }
+        };
+        return from(this.requestApiHelper.post(`dmn-service/v1/dmn/execute`, requestOptions));
+    }
+
+    create(model: Partial<T>, containerId?: string): Observable<T> {
```
Comment (popovicsandras) [inline projects/ama-sdk/src/lib/api-implementations/acm-api/dmn-api.ts]: Besides the validation, every method here seems like just dummy methods. I think it should be part of the DecisionTableApi, where we override the validation, the same way as we did for the ProcessApi.

File: projects/ama-sdk/src/lib/api-implementations/acm-api/model-variations/dmn-api-variation.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+import { Injectable } from '@angular/core';
+import { ModelApiVariation } from '../model-api';
+ import { DecisionTableContent, Model } from 'ama-sdk/src/lib/api/types';
+
+@Injectable()
+export class DmnApiVariation<M extends Model, C extends DecisionTableContent> implements ModelApiVariation<M, C> {
+
+    readonly contentType: string = 'application/json';
```
Comment (popovicsandras) [inline projects/ama-sdk/src/lib/api-implementations/acm-api/model-variations/dmn-api-variation.ts]: Although I don't think this file is even needed, but I think the dmn files' content is xml, not json, isn't it?

**Comment by dhrn** [comment]
#211

**Pull request closed without merging**